src/server/src/main/java/io/cassandrareaper/storage/operations/CassandraOperationsDao.java

```diff
@@ -1,92 +1,81 @@
 
-  public CassandraOperationsDao(Session session) {
+  public CassandraOperationsDao(CqlSession session) {
     this.session = session;
     prepareOperationsStatements();
   }
 
   private void prepareOperationsStatements() {
-    insertOperationsPrepStmt = session.prepare(
-        "INSERT INTO node_operations(cluster, type, time_bucket, host, ts, data) "
+    insertOperationsPrepStmt =
+        session.prepare("INSERT INTO node_operations(cluster, type, time_bucket, host, ts, data) "
             + "values(?,?,?,?,?,?)");
 
-    listOperationsForNodePrepStmt = session.prepare(
-        "SELECT cluster, type, time_bucket, host, ts, data FROM node_operations "
+    listOperationsForNodePrepStmt =
+        session.prepare("SELECT cluster, type, time_bucket, host, ts, data FROM node_operations "
             + "WHERE cluster = ? AND type = ? and time_bucket = ? and host = ? LIMIT 1");
   }
 
-  public void storeOperations(String clusterName, OpType operationType, String host, String operationsJson) {
-    session.executeAsync(
-        insertOperationsPrepStmt.bind(
-            clusterName,
-            operationType.getName(),
-            DateTime.now().toString(TIME_BUCKET_FORMATTER),
-            host,
-            DateTime.now().toDate(),
-            operationsJson));
+  public void storeOperations(String clusterName, OpType operationType, String host,
+      String operationsJson) {
+    session.executeAsync(insertOperationsPrepStmt.bind(clusterName, operationType.getName(),
+        DateTime.now().toString(TIME_BUCKET_FORMATTER), host, Instant.now(), operationsJson));
   }
 
   public String listOperations(String clusterName, OpType operationType, String host) {
-    List<ResultSetFuture> futures = Lists.newArrayList();
-    futures.add(session.executeAsync(
-        listOperationsForNodePrepStmt.bind(
-            clusterName, operationType.getName(), DateTime.now().toString(TIME_BUCKET_FORMATTER), host)));
-    futures.add(session.executeAsync(
-        listOperationsForNodePrepStmt.bind(
-            clusterName,
-            operationType.getName(),
-            DateTime.now().minusMinutes(1).toString(TIME_BUCKET_FORMATTER),
-            host)));
-    for (ResultSetFuture future : futures) {
-      ResultSet operations = future.getUninterruptibly();
-      for (Row row : operations) {
-        return row.getString("data");
+    List<CompletionStage<AsyncResultSet>> futures = Lists.newArrayList();
+    futures.add(session.executeAsync(listOperationsForNodePrepStmt.bind(clusterName,
+        operationType.getName(), DateTime.now().toString(TIME_BUCKET_FORMATTER), host)));
+    futures.add(session
+        .executeAsync(listOperationsForNodePrepStmt.bind(clusterName, operationType.getName(),
+            DateTime.now().minusMinutes(1).toString(TIME_BUCKET_FORMATTER), host)));
+    for (CompletionStage<AsyncResultSet> future : futures) {
+      AsyncResultSet results = future.toCompletableFuture().join();
+      while (true) {
```

Comment:
This confuses me, why do we need while(true) ? Shouldn't CompletableFuture.join() block the execution until the resultSet is available? That's how the CompletableFuture should work.

Reply:
right, the while loop can probably be removed

Reply:
actually it's needed it seems with the new paging model of the API.
But in this specific case, we're not going through pages and returning the first row we find, so we should be ok.

Reply:
Yeah, that was different from the other examples, that's why it caught my eye.